resources/aws_ec2.go

```diff
@@ -650,7 +650,7 @@ func (obj *AwsEc2Res) snsWatch() error {
 			}
 		case msg, ok := <-obj.awsChan:
 			if !ok {
-				return *exit
```

Comment:
it's true that this is a bug, but why is it an error instead of a nil return?

same question for the longPollWatch too...

Reply:
I could ignore the channel close, but because the channel is only supposed to be closed when the sender returns, this case should never happen. IOW, this will only come up if something went wrong and the channel was closed early, possibly leading to missed events. Is that not a good enough reason to return an error?

Reply:
@purpleidea I now see the error of my ways. :P Fixed now.

Reply:
Errors are ok. I make many of them.